test/e2e/dhcp_overrides_test.go

```diff
@@ -0,0 +1,121 @@
+	"github.com/vmware/govmomi/vim25/mo"
+	"gopkg.in/yaml.v2"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/cluster-api/test/framework/clusterctl"
+	"sigs.k8s.io/cluster-api/util"
+
+	types "sigs.k8s.io/cluster-api-provider-vsphere/apis/v1beta1"
+)
+
+type GuestInfoMetadata struct {
+	Network Network `yaml:"network"`
+}
+
+type Network struct {
+	Ethernets map[string]Ethernet `yaml:"ethernets"`
+}
+
+type Ethernet struct {
+	DHCP4Overrides DHCPOverrides `yaml:"dhcp4-overrides"`
+}
+
+type DHCPOverrides struct {
+	SendHostname *bool `yaml:"send-hostname"`
+}
+
+var _ = Describe("DHCPOverrides configuration test", func() {
+	When("Creating a cluster with DHCPOverrides configured", func() {
+		It("Only configures the network with the provided nameservers", func() {
+			specName := "dhcp-overrides"
+			namespace := setupSpecNamespace(specName)
+			clusterName := fmt.Sprintf("%s-%s", specName, util.RandomString(6))
+			clusterResources := new(clusterctl.ApplyClusterTemplateAndWaitResult)
+
+			By("Creating a workload cluster")
+			clusterctl.ApplyClusterTemplateAndWait(ctx, clusterctl.ApplyClusterTemplateAndWaitInput{
+				ClusterProxy: bootstrapClusterProxy,
+				ConfigCluster: clusterctl.ConfigClusterInput{
+					LogFolder:                filepath.Join(artifactFolder, "clusters", bootstrapClusterProxy.GetName()),
+					ClusterctlConfigPath:     clusterctlConfigPath,
+					KubeconfigPath:           bootstrapClusterProxy.GetKubeconfigPath(),
+					InfrastructureProvider:   clusterctl.DefaultInfrastructureProvider,
+					Flavor:                   "dhcp-overrides",
+					Namespace:                namespace.Name,
+					ClusterName:              clusterName,
+					KubernetesVersion:        e2eConfig.GetVariable(KubernetesVersion),
+					ControlPlaneMachineCount: pointer.Int64Ptr(1),
+					WorkerMachineCount:       pointer.Int64Ptr(1),
+				},
+				WaitForClusterIntervals:      e2eConfig.GetIntervals(specName, "wait-cluster"),
+				WaitForControlPlaneIntervals: e2eConfig.GetIntervals(specName, "wait-control-plane"),
+				WaitForMachineDeployments:    e2eConfig.GetIntervals(specName, "wait-worker-nodes"),
+			}, clusterResources)
+
+			list := getVSphereVMsForCluster(clusterName, namespace.Name)
+			for _, vm := range list.Items {
+				metadata := getVMMetadata(vm)
+				guestInfoMetadata := &GuestInfoMetadata{}
+				err := yaml.Unmarshal(metadata, guestInfoMetadata)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(guestInfoMetadata.Network.Ethernets["id0"].DHCP4Overrides.SendHostname).NotTo(BeNil())
+				Expect(*guestInfoMetadata.Network.Ethernets["id0"].DHCP4Overrides.SendHostname).To(BeFalse())
+			}
+		})
+	})
+})
+
+func getVMMetadata(vm types.VSphereVM) []byte {
+	finder := find.NewFinder(vsphereClient.Client, false)
+	dc, err := finder.Datacenter(ctx, vm.Spec.Datacenter)
+	Expect(err).NotTo(HaveOccurred())
+	finder.SetDatacenter(dc)
+
+	vmObj, err := finder.VirtualMachine(ctx, fmt.Sprintf("/%s/vm/%s/%s", vm.Spec.Datacenter, vm.Spec.Folder, vm.Name))
```

Comment:
```suggestion
	vmObj, err := finder.VirtualMachine(ctx, vm.Name)
```
should be enough to find the VM.

Reply:
Updated, thanks!